gateway/src/main/kotlin/com/makarytskyi/gateway/mapper/OrderMapper.kt

```diff
@@ -0,0 +1,113 @@
+        .setStartDate(dateToTimestamp(from))
+        .setEndDate(dateToTimestamp(to))
+        .build()
+
+    @SuppressWarnings("TooGenericExceptionThrown")
+    fun CreateOrderResponse.toDto() =
+        when (responseCase) {
+            CreateOrderResponse.ResponseCase.SUCCESS -> success.order.toDto()
+            CreateOrderResponse.ResponseCase.FAILURE -> failure.throwException()
+            CreateOrderResponse.ResponseCase.RESPONSE_NOT_SET -> throw RuntimeException(failure.message)
+        }
+
+    @SuppressWarnings("TooGenericExceptionThrown")
+    fun GetByIdOrderResponse.toDto() =
+        when (responseCase) {
+            GetByIdOrderResponse.ResponseCase.SUCCESS -> success.order.toDto()
+            GetByIdOrderResponse.ResponseCase.FAILURE -> failure.throwException()
+            GetByIdOrderResponse.ResponseCase.RESPONSE_NOT_SET -> throw RuntimeException(failure.message)
+        }
+
+    @SuppressWarnings("TooGenericExceptionThrown")
+    fun UpdateOrderResponse.toDto(): OrderResponseDto =
+        when (responseCase) {
+            UpdateOrderResponse.ResponseCase.SUCCESS -> success.order.toDto()
+            UpdateOrderResponse.ResponseCase.FAILURE -> failure.throwException()
+            UpdateOrderResponse.ResponseCase.RESPONSE_NOT_SET -> throw RuntimeException(failure.message)
+        }
+
+    @SuppressWarnings("TooGenericExceptionThrown")
+    fun FindAllOrdersResponse.toDto(): List<AggregatedOrderResponseDto> =
+        when (responseCase) {
+            FindAllOrdersResponse.ResponseCase.SUCCESS -> success.ordersList.stream().map { it.toDto() }.toList()
+            else -> throw RuntimeException(failure.message)
+        }
+
+    @SuppressWarnings("TooGenericExceptionThrown")
+    private fun CreateOrderResponse.Failure.throwException(): Nothing =
+        when (errorCase) {
+            CreateOrderResponse.Failure.ErrorCase.NOT_FOUND -> throw NotFoundException(message)
+            CreateOrderResponse.Failure.ErrorCase.ILLEGAL_ARGUMENT -> throw IllegalArgumentException(message)
+            CreateOrderResponse.Failure.ErrorCase.ERROR_NOT_SET -> throw RuntimeException(message)
+        }
+
+    @SuppressWarnings("TooGenericExceptionThrown")
+    private fun UpdateOrderResponse.Failure.throwException(): Nothing =
+        when (errorCase) {
+            UpdateOrderResponse.Failure.ErrorCase.NOT_FOUND -> throw NotFoundException(message)
+            UpdateOrderResponse.Failure.ErrorCase.ILLEGAL_ARGUMENT -> throw IllegalArgumentException(message)
+            UpdateOrderResponse.Failure.ErrorCase.ERROR_NOT_SET -> throw RuntimeException(message)
+        }
+
+    @SuppressWarnings("TooGenericExceptionThrown")
+    private fun GetByIdOrderResponse.Failure.throwException(): Nothing =
+        when (errorCase) {
+            GetByIdOrderResponse.Failure.ErrorCase.NOT_FOUND -> throw NotFoundException(message)
+            GetByIdOrderResponse.Failure.ErrorCase.ERROR_NOT_SET -> throw RuntimeException(message)
```

Comment:
Why runtime exception?

Reply:
Because, ERROR_NOT_SET represents other error cases and RuntimeException is general class for exceptions.

Reply:
You have suppressed `TooGenericExceptionThrown`. 
Why throwing RuntimeException is bad practice? Because it's unclear how to handle the exception. It's too generic

Reply:
So, custom exception (e.g. UnspecifiedServiceException) can be used here.